BasicSteps/SweepFileParameterStep.cs

```diff
@@ -0,0 +1,137 @@
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+
+namespace OpenTap.Plugins.BasicSteps
+{
+    [AllowAnyChild]
+    [Display("Sweep Parameter From File", "Sweeps the parameters based on a table file format. For example CSV using the CSV plugin..", "Flow Control")]
+    public class SweepFileParameterStep : SweepParameterStepBase, ISerializeNotifyAdditionalTypesUsed
+    {
+
+        [FilePath]
+        [Display("File","File containing the swept values.")]
+        public string SweepValues { get; set; }
+
+        /// <summary>
+        /// This property declares to the Resource Manager which resources are declared by this test step. 
+        /// </summary>
+        [AnnotationIgnore]
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        [Browsable(false)]
+        public IEnumerable<IResource> Resources
+        {
+            get
+            {
+                var table = LoadTable().Item1;
+                var parameters = SelectedParameters;
+                if (table != null)
+                {
+                    foreach (var row in table)
+                    {
+                        for (int i = 0; i < row.Length; i++)
+                        {
+                            var cell = row[i];
+                            var p = parameters.ElementAt(i);
+                            if (p != null && p.TypeDescriptor.DescendsTo(typeof(IResource)))
+                            {
+                                if(StringConvertProvider.TryFromString(cell, p.TypeDescriptor, this, out var r) && r is IResource r2)
+                                {
+                                    yield return r2;
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+            
+        public SweepFileParameterStep()
+        {
+            Name = "Sweep {Parameters}";
+            Rules.Add(() => File.Exists(SweepValues), "Sweep values does not exist.", nameof(SweepValues));
+        }
+        
+        public override void PrePlanRun()
+        {
+            base.PrePlanRun();
+
+            if (SelectedParameters.Count <= 0)
+                throw new InvalidOperationException("No parameters selected to sweep");
+        }
+        
+        (string[][], ITableImport) LoadTable()
+        {
+            var importers = TypeData.GetDerivedTypes<ITableImport>();
+            
+            foreach (var importType in importers)
+            {
+                var importer = (ITableImport)importType.CreateInstance();
+                if(Path.GetExtension(SweepValues).Equals(importer.Extension, StringComparison.OrdinalIgnoreCase))
+                {
+                    return (importer.ImportTableValues(SweepValues), importer);
+                }
+            }
```

Comment:
I was thinking here that checking the extension is extremely specific to tabular data formats. I guess that makes sense since this is step sweeps a file, and is based on the existing CSV importer. 

But this got me thinking that maybe this should be a `Sweep From Source` instead. Then each importer could decide if they know how to interpret the source. E.g. an SQLite importer could accept a select statement as a source.

Maybe this idea is a bit too crazy. What do you think?

Reply:
If we do such an implementation, the user should probably pick a source instead of magically trying to detect the source.